core/types/block.libevm.go

```diff
@@ -108,3 +109,93 @@ func (*NOOPHeaderHooks) DecodeRLP(h *Header, s *rlp.Stream) error {
 	type withoutMethods Header
 	return s.Decode((*withoutMethods)(h))
 }
+
+var (
+	_ interface {
+		rlp.Encoder
+		rlp.Decoder
+	} = (*Body)(nil)
+
+	// The implementations of [Body.EncodeRLP] and [Body.DecodeRLP] make
+	// assumptions about the struct fields, which we lock in here as a change
```

Comment:
nit I suppose we are looking at the fields order as well right? Just mentioning it on top:

```suggestion
	// assumptions about the struct fields and their order, which we lock in here as a change
```